### PR TITLE
Initial polish support

### PR DESCRIPTION
**Pull request by rafaljanicki**
Issue #107

image

### CODE REVIEW

**Comment by devs-mycroft** [comment]
Hello, @rafaljanicki, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

**Comment by rafaljanicki** [comment]
I've just signed the agreement and resolved the conflicts. I do remember about squashing the commits, but let me do that after the review :)

**Comment by ChanceNCounter** [comment]
I thought I'd commented, but I see now that I haven't.

This is much appreciated, and will be reviewed soon. There's a refactor coming in the next few days which will obviate a lot of this boilerplate, so we're holding off on code merging for a little bit.

Sorry for the delay letting you know what's going on. Somebody will review this just as soon as it can be diffed against the "new normal."

It should *only* affect the boilerplate, and maybe require us to change some of your function parameters' names.

**Comment by rafaljanicki** [comment]
Thanks for the update! I'll watch the updates then

**Comment by ChanceNCounter** [comment]
6 months later, I'm finally ready to review all the stale PRs I blocked with that refactor. You're first in line. It'll be interspersed with my day job throughout today and tomorrow. Sorry, and thanks, and etc!

**Comment by rafaljanicki** [comment]
Oh God, I lost all hope already ;) I've just pushed a commit which modifies the code to match 3.0.0. There's only one place which you won't like - `nice_duration` in `format.py`. This is because I had to modify this method quite a lot hence it's separated (polish language rules forced me to do so if I wanted to make it nice) and this method is not localized. I'll leave it up to you what to do @ChanceNCounter

**Comment by ChanceNCounter** [comment]
Oh, thank you! I was expecting to do that, it's much appreciated. Checking it out now :)

**Comment by ChanceNCounter** [comment]
Ah, I see what you mean. The solution here is to make Lingua Franca treat it as a localized function. I've thrown something together, which I'll push now for your consideration.

Basically, the wrapper takes a parameter that makes it fall back on the wrapped function itself, if a specified error is raised: `@localized_function(run_own_code_on: List: Exception)`

The function localizer has a bit that raises `FunctionNotLocalizedError` if it can't find the function. If, for a kludge, we have `nice_duration_pl` raise that exception if speech=False, everything fires under the right conditions.

**Comment by rafaljanicki** [comment]
It looks great, thank you! :)

**Comment by ChanceNCounter** [comment]
Wellp, nothing jumps out at me. All the tests pass, and the module would (deliberately) break if you'd only partially registered the language, so you've definitely done that correctly. Function signatures look right.

I don't speak Polish, and neither do the other maintainers here, so we really have no way of knowing if there's a swear word in there ;) but you don't seem like a troll, so that's not too concerning.

Would you like to do the squash? Feel free to roll my commit back into yours, for ease of rebasing.

File: lingua_franca/lang/format_pl.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+            return res
+
+        result += _short_scale(num)
+
+    # deal with scientific notation unpronounceable as number
+    if not result and "e" in str(num):
+        return pronounce_number_pl(num, places, short_scale, scientific=True)
+    # Deal with fractional part
+    elif not num == int(num) and places > 0:
+        if abs(num) < 1.0 and (result is "minus " or not result):
```
Comment (ChanceNCounter) [inline lingua_franca/lang/format_pl.py:225]: I apologize for failing to notice this before you rebased. This `is` raises a SyntaxWarning, fixed in the other files some time after you forked.

Easy fix, `if abs(num) < 1.0 and (result == "minus " or not result):`

`git commit --amend` should be okay, given that you're force-pushing regardless.

**Comment by rafaljanicki** [comment]
@ChanceNCounter I've just force pushed squashed commit with a nice commit message and also included your request from the comment (change "is" to "=="). I hope it's all good now :) 

Thanks!

**Comment by ChanceNCounter** [comment]
Yep, afaik that was the last thing! Just waiting for the CI system to recover from whatever is keeping it from running. Once green checkmarks show up, j'approuve

**Comment by ChanceNCounter** [comment]
At last, Travis plays ball.

Thank you, @rafaljanicki, for your immense contribution. New languages are the very biggest deal around here!

**Pull request merged**